src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1220,6 +1223,9 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                     pResult->thisTransform = CORINFO_THIS_TRANSFORM.CORINFO_NO_THIS_TRANSFORM;
 
                     exactType = directMethod.OwningType;
+
+                    if (!directMethod.IsRuntimeDeterminedExactMethod)
```

Comment:
The condition in the `if` is always true. I'd get rid of it. I think you wanted to filter out canonical forms, but I don't see them filtered out in the scanner either and it should be okay that way. We could potentially change the implementation of `NoteOverridingMethod` to grab the typical method definitions of things and report _that_ (that way it really doesn't matter what we report).

The difference between canonical forms and runtime determined forms is this (this took me a while to really understand this so maybe I won't be able to explain but I'll still try):

* Canonical form of a method (in principle) is a form where every reference type in the instantiation got replaced with `__Canon`. There can be multiple canonical forms for a single method definition (for example, `Foo<__Canon>`, `Foo<SomeGenericValueType<__Canon>>`). But each canonical form has its own physical method body with processor instructions.
* A runtime determined form has a close relationship to canonical form. It's a canonical method with extra information about what should be in the `__Canon`.

For example:

```csharp
static void Foo<T>() { }
static void Bar<T, U>()
{
    Foo<T>();
    Foo<U>();
}
```

If we're compiling the method body of `Bar<__Canon, __Canon>()` (canonical form) we need to somehow represent the destination of each of the call. This is where runtime detemined forms come into play: the first call calls `Foo<T___Canon>`, the second calls `Foo<U___Canon>` (the runtime determined form). These forms basically capture the physical method body, with extra information about where to grab the generic context from. As far as RyuJIT is concerned, it's calling `Foo<__Canon>` with some magic cookie (generic context) we made from somewhere. But the runtime determined form lets us figure out what the magic cookie is.